docs/README.md

```diff
@@ -1,107 +1,67 @@
+{
+  test: /\.js$/,
+  loaders: ['react-hot-loader/webpack', 'babel'],
+  include: path.join(__dirname, '..', '..', 'src')
+}
 ```
 
-Note the second parameter: `makeHot` needs some way to distinguish components of same type inside on module. By default, it uses `displayName` of given component class, but in case of dynamically generated classes (or if you're not using JSX), you have to provide it yourself.
+- 'react-hot-loader/patch' should be placed at the top of the `entry` section in your Webpack config.  An error will occur if any code runs before `react-hot-loader/patch` has, so put it in the first position.
 
-### Manual mode (experimental)
+- `<AppContainer>` - AppContainer is a component that handles module reloading, as well as error handling.  The root component of your app should be nested in AppContainer as a child.  When in production, AppContainer is automatically disabled, and simply returns its children.
 
-You can now use `react-hot?manual` instead of `react-hot` in Webpack config to turn on manual mode. In manual mode, “accepting” hot updates is up to you; modules won't accept themselves automatically. This can be used, for example, to put reloading logic on very top of the application and [hot-reload routes as well as components](https://github.com/rackt/react-router/pull/606#issuecomment-66936975). It will also work better when you have a lot of modules that export component-generating functions because updates will propagate to the top. (Don't worry if you don't understand this; it's just something experimental you might want to try to integrate hot reloading deeper into your app.)
-
-### Usage with external React
-
-If you're using external standalone React bundle instead of NPM package, Hot Loader will fail because it relies on `react/lib/ReactMount` which is not exposed in precompiled React. It needs `ReactMount` to keep track of mounted React component instances on the page. However, you can supply your own root instance provider:
+- React Hot Loader 3 does not hide the hot module replacement API, so the following needs to be added below wherever you call `ReactDOM.render` in your app:
 
 ```js
-// Your app's index.js
-
-var React = require('react'),
-    router = require('./router');
-
-var rootInstance = null;
-
-router.run(function (Handler, state) {
-  rootInstance = React.render(<Handler />, document.body);
-});
-
+import React from 'react'
+import ReactDOM from 'react-dom'
+import { AppContainer } from 'react-hot-loader'
+import App from './containers/App'
+
+ReactDOM.render(
+  <AppContainer>
+    <App/>
+  </AppContainer>,
+  document.getElementById('root')
+);
+
+// Hot Module Replacement API
 if (module.hot) {
-  require('react-hot-loader/Injection').RootInstanceProvider.injectProvider({
-    getRootInstances: function () {
-      // Help React Hot Loader figure out the root component instances on the page:
-      return [rootInstance];
-    }
+  module.hot.accept('./containers/App', () => {
+    render(
```

Comment:
Do we need
```
const NextApp = require('./containers/App').default
```
here?

I'm in a twilight zone ... it works without `require()` in one project, but it does not work at all in the other, no matter if I use `require()` or not.

Can anyone explain when it will work without?